sim/core/character.go

```diff
@@ -183,6 +186,31 @@ func NewCharacter(party *Party, partyIndex int, player *proto.Player) Character
 		character.enableItemSwap(player.ItemSwap, character.DefaultMeleeCritMultiplier(), character.DefaultMeleeCritMultiplier(), 0)
 	}
 
+	switch character.Spec {
+	case proto.Spec_SpecBloodDeathKnight, proto.Spec_SpecFeralDruid, proto.Spec_SpecProtectionPaladin, proto.Spec_SpecProtectionWarrior:
+		character.PrimaryStat = proto.PrimaryStat_PrimaryStatStamina
+	case proto.Spec_SpecFrostDeathKnight, proto.Spec_SpecUnholyDeathKnight, proto.Spec_SpecRetributionPaladin, proto.Spec_SpecArmsWarrior, proto.Spec_SpecFuryWarrior:
+		character.PrimaryStat = proto.PrimaryStat_PrimaryStatStrength
+	case proto.Spec_SpecBalanceDruid, proto.Spec_SpecRestorationDruid, proto.Spec_SpecArcaneMage, proto.Spec_SpecFireMage, proto.Spec_SpecFrostMage, proto.Spec_SpecHolyPaladin,
+		proto.Spec_SpecDisciplinePriest, proto.Spec_SpecHolyPriest, proto.Spec_SpecShadowPriest, proto.Spec_SpecElementalShaman, proto.Spec_SpecRestorationShaman,
+		proto.Spec_SpecAfflictionWarlock, proto.Spec_SpecDemonologyWarlock, proto.Spec_SpecDestructionWarlock:
+		character.PrimaryStat = proto.PrimaryStat_PrimaryStatIntellect
+	case proto.Spec_SpecBeastMasteryHunter, proto.Spec_SpecMarksmanshipHunter, proto.Spec_SpecSurvivalHunter, proto.Spec_SpecAssassinationRogue,
+		proto.Spec_SpecCombatRogue, proto.Spec_SpecSubtletyRogue, proto.Spec_SpecEnhancementShaman:
+		character.PrimaryStat = proto.PrimaryStat_PrimaryStatAgility
+	}
+
+	switch character.Class {
+	case proto.Class_ClassMage, proto.Class_ClassPriest, proto.Class_ClassWarlock:
+		character.ArmorSpecialization = proto.ArmorSpecialization_ArmorSpecializationCloth
+	case proto.Class_ClassDruid, proto.Class_ClassRogue:
+		character.ArmorSpecialization = proto.ArmorSpecialization_ArmorSpecializationLeather
+	case proto.Class_ClassHunter, proto.Class_ClassShaman:
+		character.ArmorSpecialization = proto.ArmorSpecialization_ArmorSpecializationMail
+	case proto.Class_ClassDeathKnight, proto.Class_ClassPaladin, proto.Class_ClassWarrior:
+		character.ArmorSpecialization = proto.ArmorSpecialization_ArmorSpecializationPlate
+	}
```

Comment:
I dont like the core having this knowledge about specs. We should instead make the armor specialization be enabled from the specs .go files with something like
 `dk.EnableArmorSpecialization(stats.StatStrength, proto.ArmorType_ArmorTypePlate)`